presentation/src/main/java/com/martgant/tolkienquizapp/QuestionFragment.kt

```diff
@@ -0,0 +1,76 @@
+package com.martgant.tolkienquizapp
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.StringRes
+import androidx.fragment.app.Fragment
+import com.martgant.tolkienquizapp.databinding.FragmentQuestionBinding
+import kotlin.random.Random
+
+class QuestionFragment : Fragment() {
+
+    private var _binding: FragmentQuestionBinding? = null
+    private val binding get() = _binding!!
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+
+        _binding = FragmentQuestionBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        val (questionRes, answer1Res, answer2Res) = getQuestionProps()
+        binding.questionDescription.text = context?.getString(questionRes)
+        binding.answer1.text = context?.getString(answer1Res)
+        binding.answer2.text = context?.getString(answer2Res)
+    }
+
+    private fun getQuestionProps(): QuestionProps {
+        val questionNumber = Random.nextInt(1, 5)
+
+        val questionFormat = null
+        val answerCorrectFormat = null
+        val answerIncorrectFormat = null
+        return QuestionProps(
+            questionRes = resources.getIdentifier(
+                questionFormat.format(questionNumber),
+                "string",
+                activity?.packageName
+            ),
+            answer1Res = resources.getIdentifier(
+                answerCorrectFormat.format(questionNumber),
+                "string",
+                activity?.packageName
+            ),
+            answer2Res = resources.getIdentifier(
+                answerIncorrectFormat.format(questionNumber, 1),
+                "string",
+                activity?.packageName
+            )
+        )
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    private data class QuestionProps(
+        @StringRes val questionRes: Int,
+        @StringRes val answer1Res: Int,
+        @StringRes val answer2Res: Int
+    )
+
+    companion object {
+        private const val QUESTION_FORMAT = "question_%d"
+        private const val ANSWER_CORRECT_FORMAT = "question_%d_answer_correct"
+        private const val ANSWER_INCORRECT_FORMAT = "question_%d_answer_incorrect_%d"
+    }
```

Comment:
The convention is to place companion object at the top of the class. Also if all the fields in companion object are private (like here) you can make companion object private itself and then you can drop `private` for the fields. So:
```
 private companion object {
       const val QUESTION_FORMAT = "question_%d"
       const val ANSWER_CORRECT_FORMAT = "question_%d_answer_correct"
       const val ANSWER_INCORRECT_FORMAT = "question_%d_answer_incorrect_%d"
    }
```